pkg/scheduler/cache/cache.go

```diff
@@ -240,8 +242,112 @@ func (cache *schedulerCache) addPod(pod *v1.Pod) {
 	n.AddPod(pod)
 }
 
+// this function expects valid pod, and valid, non-empty resizeRequestAnnotation json string
+func (cache *schedulerCache) getPodResizeRequirements(pod *v1.Pod, resizeRequestAnnotation string) (map[string]v1.Container, *Resource, error) {
+	var resizeContainers []v1.Container
+
+	if err := json.Unmarshal([]byte(resizeRequestAnnotation), &resizeContainers); err != nil {
+		return nil, nil, err
+	}
+
+	resizeContainersMap := make(map[string]v1.Container)
+	for _, resizeContainer := range resizeContainers {
+		resizeContainersMap[resizeContainer.Name] = resizeContainer
+	}
+
+	podResource := &Resource{}
+	for _, container := range pod.Spec.Containers {
+		containerResourcesRequests := v1.ResourceList {
+							v1.ResourceCPU:    container.Resources.Requests[v1.ResourceCPU],
+							v1.ResourceMemory: container.Resources.Requests[v1.ResourceMemory],
+						}
+		if resizeContainer, ok := resizeContainersMap[container.Name]; ok {
+			for k, v := range resizeContainer.Resources.Requests {
+				containerResourcesRequests[k] = v
+			}
+		}
+		podResource.Add(containerResourcesRequests)
+	}
+
+	return resizeContainersMap, podResource, nil
+}
+
+func (cache *schedulerCache) processPodResourcesResizeRequest(oldPod, newPod *v1.Pod) error {
+	node, ok := cache.nodes[newPod.Spec.NodeName]
+	if !ok {
+		errMsg := fmt.Sprintf("Node %s not found for pod %s", newPod.Spec.NodeName, newPod.Name)
+		glog.Errorf(errMsg)
+		return fmt.Errorf(errMsg)
+	}
+
+	resizeResourcesPolicy := api.ResizePolicyInPlacePreferred
+	if _, ok := newPod.ObjectMeta.Annotations[api.AnnotationResizeResourcesPolicy]; ok {
+		resizeResourcesPolicy = api.ResizePolicy(newPod.ObjectMeta.Annotations[api.AnnotationResizeResourcesPolicy])
+	}
+
+	if resizeRequestAnnotation, ok := newPod.ObjectMeta.Annotations[api.AnnotationResizeResources]; ok {
+		if resizeRequestAnnotation == "" {
+			return nil
```

Comment:
please consider adding a log here. it'd be interesting to know when this case is happening.

Reply:
This happens because once I'm done with processing the annotation in factory.go, I set it to "". The other option would be to remove the annotation completely once handled.

Reply:
I'd remove it in this case. Having an annotation of an empty string seems a little confusing.

Reply:
removed.